pkg/config/derive.go

```diff
@@ -139,3 +139,87 @@ func (derived) StatsdArgs(cs *api.OpenShiftManagedCluster) ([]interface{}, error
 		"-SourceRoleInstance", "OSA",
 	}, nil
 }
+
+// MaxDataDisksPerVM is a stopgap until k8s 1.12.  It requires that a cluster
+// has only one compute AgentPoolProfile and that no infra VM will require more
+// mounted disks than the maximum number allowed by the compute agent pool.
+// https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes
+func (derived) MaxDataDisksPerVM(cs *api.OpenShiftManagedCluster) (string, error) {
+	var app *api.AgentPoolProfile
+	for i := range cs.Properties.AgentPoolProfiles {
+		if cs.Properties.AgentPoolProfiles[i].Role != api.AgentPoolProfileRoleCompute {
+			continue
+		}
+
+		if app != nil {
+			return "", fmt.Errorf("found multiple compute agentPoolProfiles")
+		}
+
+		app = &cs.Properties.AgentPoolProfiles[i]
```

Comment:
you could break here

Reply:
no, because the if test above would never succeed in that case